set.mm

```diff
@@ -33556,19 +33557,47 @@ general is seen either by substitution (when the variable ` V ` has no
 
   ${
     $d x ph $.  $d x ch $.  $d x A $.
-    elabd2.1 $e |- ( ph -> A e. V ) $.
-    elabd2.2 $e |- ( ( ph /\ x = A ) -> ( ps <-> ch ) ) $.
+    elabd2.ex $e |- ( ph -> A e. V ) $.
+    elabd2.eq $e |- ( ph -> B = { x | ps } ) $.
+    elabd2.is $e |- ( ( ph /\ x = A ) -> ( ps <-> ch ) ) $.
     $( Membership in a class abstraction, using implicit substitution.
        Deduction version of ~ elab .  (Contributed by Gino Giotto,
-       12-Oct-2024.) $)
-    elabd2 $p |- ( ph -> ( A e. { x | ps } <-> ch ) ) $=
-      ( wcel cab wb wa cv wceq wi wal elab6g adantl wex elisset pm5.74da albidv
-      19.23v bitrdi pm5.5 sylan9bb sylan2 bitrd mpdan ) AEFIZEBDJIZCKGAUJLUKDME
-      NZBOZDPZCUJUKUNKABDEFQRUJAULDSZUNCKDEFTAUNUOCOZUOCAUNULCOZDPUPAUMUQDAULBC
-      HUAUBULCDUCUDUOCUEUFUGUHUI $.
+       12-Oct-2024.)  (Revised by BJ, 16-Oct-2024.) $)
+    elabd2 $p |- ( ph -> ( A e. B <-> ch ) ) $=
+      ( wcel wb wa cv wceq wi wal cab eleq2d sylan9bb elab6g wex elisset albidv
+      pm5.74da 19.23v bitrdi pm5.5 sylan2 bitrd mpdan ) AEGKZEFKZCLHAULMUMDNEOZ
+      BPZDQZCAUMEBDRZKULUPAFUQEISBDEGUATULAUNDUBZUPCLDEGUCAUPURCPZURCAUPUNCPZDQ
+      USAUOUTDAUNBCJUEUDUNCDUFUGURCUHTUIUJUK $.
     $( $j usage 'elabd2' avoids 'ax-10' 'ax-11' 'ax-12'; $)
   $}
 
+  ${
+    $d x y ph $.  $d x y ch $.  $d x y A $.  $d B y $.
+    elabd2ALT.ex $e |- ( ph -> A e. V ) $.
+    elabd2ALT.eq $e |- ( ph -> B = { x | ps } ) $.
+    elabd2ALT.is $e |- ( ( ph /\ x = A ) -> ( ps <-> ch ) ) $.
+    $( Alternate proof of ~ elabd2 bypassing ~ elab6g .  (Contributed by BJ,
```

Comment:
What information is provided by avoiding elab6g?

Reply:
Is this enough of an explanation @GinoGiotto ? I can either
1. leave it like that
2. add what I wrote above in the comment of elabd2ALT
3. move it my mathbox as bj-elabd2ALT (though I think it can stay in Main)

Reply:
I don't see what makes this proof worth highlighting. It’s longer than the main one, it uses the same axioms and it also introduces an additional variable y by using proper substitution `[ y / x ] ps`, which is also wasteful since I am aware of past efforts to minimize dummy variables as well (elabd2 avoids it by using `A. x (x = A → ps)` directly).

If it's just an alternative proof with no clear benefits, I would prefer to keep it in a mathbox, rather than as an ALT version in main.

Reply:
Moved to my mathbox.  Of course it is longer than the proof of elabd2, else it would not be an ALT. It is shorter however than elab6g+elabd2. It does use a dummy variable y, like elab6g. The point is "bypassing elab6g; not using the `A. x ( x = y -> ph )` idiom".